doc/API.md

```diff
@@ -28,17 +32,379 @@ struct {
+
+'unconfirmedIncomingSiacoins' is the number of siacoins are entering the wallet
+according to the set of unconfirmed transactions. This number is often inflated
+by outgoing siacoins, because outputs are frequently larger than the amount
+being sent. The refund will be included in the unconfirmed incoming siacoins
+balance.
+
+'siafundBalance' is the number of siafunds available to the wallet as
+of the most recent block in the blockchain.
+
+'siacoinClaimBalance' is the number of siacoins that can be claimed from the
+siafunds as of the most recent block. Because the claim balance increases every
+time a file contract is created, it is possible that the balance will increase
+before any claim transaction is confirmed.
+
+#### /wallet/close [PUT]
+
+Function: Locks and closes the wallet, preparing for shutdown. After being
+closed, the keys are encrypted. Queries for the seed, to send siafunds, and
+related queries become unavailable. Queries concerning transaction history and
+balance are still available.
+
+Parameters: none
+
+Response: Standard.
+
+#### /wallet/history [GET]
+
+Function: Return a list of transactions related to the wallet.
+
+Parameters:
+```
+struct {
+	startHeight int
+	endHeight   int
+}
+```
+'startHeight' refers to the height of the block where transaction history
+should begin.
+
+'endHeight' refers to the height of of the block where the transaction history
+should end. If 'endHeight' is greater than the current height, all transactions
+up to and including the most recent block will be provided.
+
+Response:
+```
+struct {
+	confirmedHistory   []WalletTransaction
+	unconfirmedHistory []WalletTransaction
+}
+```
+'confirmedTransactions' lists all of the confirmed wallet transactions
+appearing between height 'start' and height 'end' (inclusive).
+
+'unconfirmedTransactions' lists all of the unconfirmed transactions.
+
+Wallet transactions are transactions that have been processed by the wallet and
+given more information, such as a confirmation height and a timestamp. Each
+wallet transaction contins information about only a single input or output. One
+network transaction with many inputs an outputs can result in many wallet
+transactions. All of the wallet transactions created by a network transaction
+are guaranteed to be consecutive in history. Wallet transactions will always be
+returned in chronological order.
+
+A wallet transaction takes the
+following form:
+```
+struct WalletTransaction {
+	TransactionID         string
+	ConfirmationHeight    int
+	ConfirmationTimestamp uint64
+
+	FundType       string
+	OutputID       string
+	RelatedAddress string
+	Value          int
+}
+```
+'TransactionID' is the id of the transaction from which the wallet transaction
+was derived. The full transaction can be obtaied by calling
+'/wallet/transaction/$(id)'
+
+'ConfirmationHeight' is the height at which the transaction was confirmed. The
+height will be set to 'uint64MAX' if the transaction has not been confirmed.
```

Comment:
well that's gonna cause problems, since in the struct its type is `int`. How about -1?